gcp/appengine/blog/content/posts/announcing-transitive-dependency-support-for-maven-pomxml-in-osv-scanner/index.md

```diff
@@ -0,0 +1,56 @@
+---
+title: "Announcing Transitive Dependency Support for Maven pom.xml in OSV-Scanner"
+date: 2024-06-20T00:00:00Z
+draft: false
+author: Xueqin Cui
+---
+
+We are excited to announce that OSV-Scanner now supports transitive dependency scanning for Maven pom.xml.
+
+This highly requested [feature](https://github.com/google/osv-scanner/issues/35) empowers you to detect vulnerabilities in both your direct and indirect dependencies, giving you a complete picture of your project's open source security. With this feature, OSV-Scanner fixes one out of two ecosystems/formats when it comes to transitive scanning capabilities.
+
+<!--more-->
+
+## How it works?
+
+OSV-Scanner uses [deps.dev’s Maven parser library](https://pkg.go.dev/deps.dev/util/maven) to generate an effective POM which includes project inheritance, importing managed dependencies and project interpolation. This may involve fetching Maven projects from Maven Central repository.
+
+OSV-Scanner then invokes [deps.dev’s resolver library](https://pkg.go.dev/deps.dev/util/resolve) to compute a dependency graph for the processed project. This graph includes all the direct and transitive dependencies of your project. Maven requirements are provided by [deps.dev API](https://docs.deps.dev/api/v3/#getrequirements).
+
+After the dependency resolution, OSV-Scanner queries the OSV database for vulnerabilities associated with these dependencies.
+
+## Using transitive dependency support
+
+This feature is enabled by default in OSV-Scanner when you scan a file or a directory.
```

Comment:
I'm assuming there's an implied "in your Maven project" or words to that effect here?

Reply:
yes I am trying to say this is enabled by default whatever use cases, "a file in your Maven project" sounds good but "a directory in your Maven project" may not be accurate since there might be projects of other ecosystems in this directory.